exercises/joseph_ramsay/ex02_oo_basics/src/shape.h

```diff
@@ -0,0 +1,18 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+
+#include <string>
+
+enum ShapeType { T_square, T_circle, T_rectangle, T_triangle };
+
+class Shape {
+
+public:
+  virtual int getNumSides() const = 0;
+  virtual double getArea() const = 0;
+  virtual double getPerimeter() const = 0;
+  virtual ShapeType getType() const = 0;
+  virtual std::string toString() const = 0;
```

Comment:
You are making use of polymorphism. You need a virtual destructor. See [here](https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors). Make sure you read about why this is needed. That is quite easy to get wrong, but is very important.